### DISCUSSION TITLE
Something went wrong error when clicking on post

### DISCUSSION THREAD

**Discussion by okpierre** [pixelfed/pixelfed#5137] [Support]
PHP: 8.1.12 / Laravel: 9.38.0 / Pixelfed: 0.11.4 https://github.com/pixelfed/pixelfed/commit/56f5facc30cfe57213985ff33c70288b9755a4f2

Error message is "Something went wrong" (see urls below)

this works: 
`https://example.com/i/web/post/483605114894977813`

this is broken:
`https://example.com/p/test/483605114894977813?fs=1`

`League\Flysystem\PathPrefixer::prefixPath(): Argument #1 ($path) must be of type string, null given, called in /pixelfed/vendor/laravel/framework/src/Illuminate/Filesystem/AwsS3V3Adapter.php on line 51 (View: /pixelfed/resources/views/status/show.blade.php)`

![pixelfederrorclick](https://user-images.githubusercontent.com/1679025/201693235-b9333b13-34e4-4e5e-a2b2-b7ccb2590481.png)

**Comment by MarkoKaartinen** [comment]
I have the same error. Using s3 (vultr object storage).

**Comment by ghost** [comment]
Same here with S3 (Wasabi) - when I am logged in, each post is displayed properly. If being logged out and visiting my profile page, each click on any photo causes the error.

Diagnostic details don't explain much here... "eyJpdiI6ImhWQ0J5eXZ0WVZ3NUxyVWxOT1VnK2c9PSIsInZhbHVlIjoialo0dXBYS2ZBdHpxd3Z5WnAzcGg1eW01QTRpOTVZbXpCOE44QzFNaTVSTXpwejg5VmNLa3VQY2psYkRSai9yaVdwOEJ2aHZiUjNtc0ZQM2NScndhd3ZzWmdDSmdPWHM4dkxwR2w2eU5vdzBMYi9zamp5M2ZkR2IzY3JURDFYajYxSEk1emJ4bi82eW5Fc2QzMWlpeFF2cnRwUlJ1L2hIOVlNR04vaXgxSWduNUpyTmw1YkZZWlBQTGd3UWZVajdhMkgwVnhuYUVWUnQ5RWh6TDVuM2NrdVhSYXNMNVJLT1k5OEYyUFlMZVlWQURsYmdIelR6dzNLMTlXZ2lWL1VNSmFFM0RGdnZzY2YxTXAyQzkzeVdMNW80THc3N1ZpdzNBMDVHUmxlOStFOW5KS1lPWFB0S3N2aWg2TlV3cWg5L0NwYTNYUy9LWTNZUi91TUFwQUIyRm1wS09zN0U5UmlOK0FXSUtHT08wcHNPUkdTUUhHSU9CZC82b3M0OXpMekZjYzBtOWJNSE5JNmNWd3ZaSWtZWUdZdz09IiwibWFjIjoiOWU1OTBkMzg2MDE4ZmE3Nzk3MDQzOTg5OGIyMmNmZDljNWM2NDA4MzIyNjBjNmQwZDhlNTExMGI5N2MwYjRkNiIsInRhZyI6IiJ9"

**Comment by ghost** [comment]
Solved this by sticking to one of the other discussions where, even if going with S3 all the way, FILESYSTEM_DRIVER must be set to the value of "local".

## S3 Configuration (Post-Installer)
PF_ENABLE_CLOUD=true
PF_LOCAL_AVATAR_TO_CLOUD=true
FILESYSTEM_DRIVER=local
FILESYSTEM_CLOUD=s3

**Comment by mrmannet** [comment]
> Solved this by sticking to one of the other discussions where, even if going with S3 all the way, FILESYSTEM_DRIVER must be set to the value of "local".

I tried this but it super broke everything else, images stopped loading completely and the system ground to a halt
